plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -227,29 +222,122 @@ func (g *genericScheduler) Preempt(pod *v1.Pod, nodeLister algorithm.NodeLister,
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	for len(nodeToVictims) > 0 {
-		node := pickOneNodeForPreemption(nodeToVictims)
-		if node == nil {
+
+	// scheduler extenders that do not support preemption
+	unPreemptableExtenders := []algorithm.SchedulerExtender{}
+
+	if len(nodeToVictims) > 0 {
+		for _, extender := range g.extenders {
+			if extender.SupportsPreemption() {
+				// Pass nodeToVictims to extender to do extra preemption logics. We can expect it returns a nodeToVictimsAfterExtender struct with:
+				//   1. Subset of the given nodes after Filter.
+				//   2. Additional victims on the nodes after preemption.
+				if nodeToVictimsAfterExtender, err := extender.ProcessPreemption(pod, nodeToVictims, g.cachedNodeInfoMap, pdbs); err != nil {
+					return nil, nil, nil, err
+				} else {
+					// Write back result after extender to nodeToVictims.
```

Comment:
It's not safe for me because we don't know how user implement `ProcessPreemption` in his extender. He may only return subset or any unexpected dataset. 

Or, we may want to make this assumption and document it clearly?

Reply:
> It's not safe for me because we don't know how user implement ProcessPreemption in his extender. He may only return subset or any unexpected dataset.

Aren't you effectively making `nodeToVictims` the same as `nodeToVictimsAfterExtender` here?

> Or, we may want to make this assumption and document it clearly?

I think we should assume that extenders work fine. Without this assumption the whole logic around extenders may fail. For example, an extender may incorrectly return an empty set of nodes for preemption and by doing this, it effectively blocks preemption. There is no way for us to detect such errors. So, we must assume that extenders work fine in order for the logic to make sense.

Reply:
Sure, will update based on this assumption.